### PR TITLE
Add Calibrated Projection

### PR DESCRIPTION
**Pull request by SangdonLim**
* `runLinking()` now supports `method = 'CP'` to perform two-dimensional calibration, for use in performing calibrated projection ([Thissen et al., 2011](https://doi.org/10.1007/s11136-011-9874-y)).
* `runLinking()` now supports `method = 'CPLA'` to perform two-dimensional calibration, for use in performing linear approximation of calibrated projection ([Thissen et al., 2015](https://doi.org/10.1007/978-3-319-19977-1_1)).
* `runRSSS()` now performs two-dimensional Lord-Wingersky recursion with numerical integration, when the output from `runLinking(method = 'CP')` is supplied.
* `runRSSS()` now performs linear approximation of calibrated projection, when the output from `runLinking(method = 'CPLA')` is supplied.

Example code using the `data_asq` dataset:

```
library(PROsetta)

calib    <- runLinking(data_asq, method = "CP")

calib$method <- "CP"
s_table      <- runRSSS(data_asq, calib, inc = .2)
table_CP     <- s_table[[1]]

calib$method <- "CPLA"
s_table      <- runRSSS(data_asq, calib, inc = .2)
table_CPLA   <- s_table[[1]]

plot(
  0, 0, type = "n", xlim = c(29, 145), ylim = c(-2, 4),
  xlab = "Scale 1 (raw score)", ylab = "Scale 2 (EAP theta)")
lines(table_CPLA$raw_1, table_CP$eap_dim2  , col = "blue", lty = 3, lwd = 2)
lines(table_CPLA$raw_1, table_CPLA$eap_dim2, col = "red" , lty = 3, lwd = 2)
legend(
  "topleft",
  c("Calibrated projection",
    "Calibrated projection (linear approximation)"),
  lty = 3, lwd = 2,
  col = c("blue", "red")
)

plot(
  0, 0, type = "n", xlim = c(29, 145), ylim = c(-2, 4),
  xlab = "Scale 1 (raw score)", ylab = "Scale 2 (EAP theta + 1 SE)")
lines(table_CPLA$raw_1, table_CP$eap_dim2   + table_CP$eap_se_dim2  , col = "blue", lty = 3, lwd = 2)
lines(table_CPLA$raw_1, table_CPLA$eap_dim2 + table_CPLA$eap_se_dim2, col = "red" , lty = 3, lwd = 2)
legend(
  "topleft",
  c("Calibrated projection",
    "Calibrated projection (linear approximation)"),
  lty = 3, lwd = 2,
  col = c("blue", "red")
)
```

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -25,7 +25,7 @@ Description: Perform scale linking to establish relationships between instrument
     that measure similar constructs according to the PROsetta Stone methodology, as in Choi, Schalet, Cook, & Cella (2014) <doi:10.1037/a0035768>.
 URL: http://prosettastone.org, https://choi-phd.github.io/PROsetta/
 BugReports: https://github.com/choi-phd/PROsetta/issues
-Imports: equate, lavaan, mirt, plink, psych, methods
+Imports: equate, lavaan, mirt, plink, psych, methods, mvnfast, mirtCAT
```
Comment (choi-phd) [inline DESCRIPTION:28]: Let's add the prob functions and replace mirtCAT

Comment (SangdonLim) [reply DESCRIPTION:28]: @choi-phd Understood. I will let you know when the updated PR is ready.

File: man/runCalibration.Rd [hunk 2]
```diff
@@ -4,11 +4,19 @@
...
+  dimensions = 1,
+  fixedpar = FALSE,
+  ignore_nonconv = FALSE,
+  ...
+)
 }
 \arguments{
 \item{data}{a \code{\linkS4class{PROsetta_data}} object. See \code{\link{loadData}} for loading a dataset.}
 
+\item{dimensions}{number of dimensions to use. Must be 1 or 2. If 1, use one underlying dimension for all instruments combined. If 2, use each dimension separately for the anchor instrument and the developing instrument. Covariance between dimensions is freely estimated. (default = `1`)}
```
Comment (choi-phd) [inline man/runCalibration.Rd:18]: why the back tick? `1`

File: R/import.R [hunk 3]
```diff
@@ -8,4 +8,6 @@
 #' @importFrom stats dnorm sd cor
 #' @importFrom grDevices pdf jpeg png tiff dev.off
 #' @importFrom graphics hist text points box legend grid lines strheight
+#' @importFrom mvnfast dmvn
+#' @importFrom mirtCAT generate.mirt_object
```
[the remark this answers is not in the document]
Comment (SangdonLim) [reply R/import.R:12]: The use of `mirtCAT::generate.mirt_object` is now removed.

File: man/runCalibration.Rd (hunk 2 above)
Comment (SangdonLim) [reply man/runCalibration.Rd:18]: @choi-phd I am using code case for all default values.

File: tests/testthat/test_runLinking.R [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  expect_equal(prod(calib$constants), -0.1620018, tolerance = 1e-06)
+  expect_equal(sum(calib$ipar_linked), 362.1947, tolerance = 1e-04)
+
+  set.seed(1)
+  calib <- runLinking(d, method = "FIXEDPAR", technical = list(NCYCLES = 1000))
+  n_anchor_items <- dim(calib$ipar_anchor)[1]
+  d <- calib$ipar_linked[1:n_anchor_items, ] - calib$ipar_anchor
+  expect_equal(sum(d), 0, tolerance = 1e-08)
+  expect_equal(sum(calib$ipar_linked), 364.7372, tolerance = 1e-04)
+
```
[the remark this answers is not in the document]
Comment (SangdonLim) [reply tests/testthat/test_runLinking.R:44]: @choi-phd Added test for CP.

File: DESCRIPTION (hunk 1 above)
Comment (SangdonLim) [reply DESCRIPTION:28]: @choi-phd Removed mirtCAT dependency.

**Pull request merged**